### PR TITLE
New Crosstalk Model

### PR DESCRIPTION
**Pull request by r-pascua**
This PR implements an improved crosstalk model, meant to better replicate crosstalk seen in H1C observations. Some of the `Simulator` code had to be modified (in particular, stuff to do with parsing the model signature and pulling parameters from the `Simulator` object as needed), but most of the changes are in the `sigchain` module.

I still need to write tests and document the new class, but I figure we could start the review process now.

Here's an example of simulated crosstalk (on top of foreground+eor visibilities) at a single time, based on the IDR2 validation sims.
example_full_array_xtalk

### CODE REVIEW

File: hera_sim/sigchain.py
Comment (steven-murray) [inline hera_sim/sigchain.py:512]: can we please use `n_copies`?

File: hera_sim/sigchain.py
Comment (steven-murray) [inline hera_sim/sigchain.py:560]: not sure if it matters, but if  `amp_decay_base` is constant, this would be faster if it was evaluated outside the function.

File: hera_sim/simulate.py
Comment (steven-murray) [inline hera_sim/simulate.py:893]: pinning a TODO

**Review by steven-murray — COMMENTED** [review]
Thanks @r-pascua, it's hard for me to grok the physics here, but the code looks good (just a couple of suggestions). Obviously still needs docs/tests. I'd also prefer if someone like @jsdillon would also review.

File: hera_sim/simulate.py
Comment (r-pascua) [reply hera_sim/simulate.py:893]: Ah, ok, the parenthetical isn't true (since I just kinda hardwired some stuff to make it work), but I still want to address the other bit. Do you think it's important enough to get this done in this PR, or should I punt this to another PR?

File: hera_sim/sigchain.py
Comment (r-pascua) [reply hera_sim/sigchain.py:560]: `amp_decay_base` is indeed constant within the scope of the `__call__` method, so I can clean this up a bit. Thanks for the tip!

**Comment by steven-murray** [comment]
OK cool thanks for the explanation! I wonder, for the "extra bits" that are attached to the main peak -- if they're not physically from the same source, should they be a different Model? The user can then choose to add either or both.

**Comment by r-pascua** [comment]
OK, sorry for the delay, but I believe this is now ready for review. I've fleshed out the docs and written a (somewhat simple) test to make sure it works. I think I want to leave the change to the `Simulator` (in particular, how it parses class `__init__` and `__call__` methods to figure out what parameters to pass) to another PR.

@steven-murray and @jsdillon, if you could give a review when you have time, then that would be great!

File: hera_sim/sigchain.py
Comment (jsdillon) [inline hera_sim/sigchain.py:558]: Clarify that this is the number of peaks at both positive and negative delays separately.

**Review by jsdillon — APPROVED** [review]
Just reviewing the new `OverAirCrossCoupling` class, which is what I think you want me to look at, I just have one tiny docstring tweak. Otherwise, looks good.

If you're feeling ambitious, I'd be really great to take the code in HERA memo 104 and apply it to a full-array simulation with this code (you can do it for a single 60-integration noise-free file, I think) to see if you get back out the same emitter position, cable delays, amplitude slope, etc. that you put in. Not sure if that's strictly necessary, but I think that'd be the ultimate check of the code.

File: hera_sim/sigchain.py
Comment (r-pascua) [reply hera_sim/sigchain.py:558]: Done, let me know if it's okay.

**Comment by r-pascua** [comment]
@jsdillon thanks for the review! I was thinking it would be good to perform the test you suggest. I'll see about running that test when I have a bit of free time. What do you think about adding a `delay_offset` parameter?

File: hera_sim/sigchain.py
Comment (jsdillon) [reply hera_sim/sigchain.py:558]: Did you push the change?

**Comment by jsdillon** [comment]
> @jsdillon thanks for the review! I was thinking it would be good to perform the test you suggest. I'll see about running that test when I have a bit of free time. What do you think about adding a `delay_offset` parameter?

I explored whether there was any good evidence for a `delay_offset` in the memo and I'm not convinced it leaving it as a free parameter made the fit appreciably better. I think we can leave it out of this simulator.

File: hera_sim/sigchain.py
Comment (r-pascua) [reply hera_sim/sigchain.py]: Oh, sorry, I guess I had a typo in the branch I pushed it to. Should be fixed now.

Comment (jsdillon) [reply hera_sim/sigchain.py]: looks good to me.

**Comment by r-pascua** [comment]
Ok, gotcha. Would you like me to run the simulation on an array before merging this in? I also need to do the typing for the new class parameters, so I won't merge just yet.

**Comment by jsdillon** [comment]
I leave that to you. I think making that a prerequisite of pulling in this PR is a somewhat selective demand for rigor. If you feel like it would build confidence you want to have, go for it. I don't think it has to be part of the unit_tests, of course, but if you do it, definitely post plots here.

File: hera_sim/simulate.py
Comment (steven-murray) [reply hera_sim/simulate.py:893]: let's postpone to another PR.

**Review by steven-murray — APPROVED** [review]
Thanks @r-pascua this looks good to me!

**Pull request merged**
